internal/cli/publish.go

```diff
@@ -179,13 +175,23 @@ func PublishCmd(ctx context.Context, outputRefs string, archs []types.Architectu
 
 	mtx := sync.Mutex{}
 
-	for _, arch := range bc.ImageConfiguration.Archs {
+	for _, arch := range archs {
 		arch := arch
-		bc := *bc
+		// working directory for this architecture
+		wd := filepath.Join(workDir, arch.ToAPK())
+		bc, err := build.New(wd, opts...)
+		if err != nil {
+			return err
+		}
+
+		// we do not generate SBOMs for each arch, only possibly for final image
+		bc.Options.SBOMFormats = []string{}
+		bc.Options.WantSBOM = false
```

Comment:
@deitch @jdolitsky Help me understand the thinking here 😃 why wouldn't we want an SBOM for each arch? cc: @kaniini

Reply:
OK, now I feel silly. I thought this comment was on #499 , not on this PR #495 . Oops. I only realized when I went to show the output with sbom-per-arch-and-index.

Thinking back, I cannot recall now why this was added. Which sounds even stranger. I cannot recall if this was something we specifically needed to do, or if it was covered elsewhere, or if this was a copy-paste logic error.

Reply:
Oh, OK, I do remember. See [this line](https://github.com/chainguard-dev/apko/pull/495/files#diff-cdd538ee9a139b124886b018f25d1dbd7f4731f27c6305847563a62a37bff775L166-L171) in the original:

```go
	// This is a hack to skip the SBOM generation during
	// image build. Will be removed when global options are a thing.
	formats := bc.Options.SBOMFormats
	wantSBOM := bc.Options.WantSBOM
	bc.Options.SBOMFormats = []string{}
	bc.Options.WantSBOM = false
```

It removes the SBoM generation for each image arch, and only restores it at the end [here](https://github.com/chainguard-dev/apko/pull/495/files#diff-cdd538ee9a139b124886b018f25d1dbd7f4731f27c6305847563a62a37bff775L265-L280) (both versions of the code, old and new):

```go
	// This is a hack to skip the SBOM generation during
	// image build. Will be removed when global options are a thing.
	formats := bc.Options.SBOMFormats
	wantSBOM := bc.Options.WantSBOM
	bc.Options.SBOMFormats = []string{}
	bc.Options.WantSBOM = false
```

Reply:
Basically, it defers building SBoM until after all images are done. I am not sure why, but it worked, and I saw no reason to change it.